cpp/marc_tools/single_marc_convert_keywords.cc

```diff
@@ -61,55 +57,46 @@ void readInGndKeywords(MARC::Reader * const marc_reader, std::unordered_map<std:
             gnd_keywords->emplace(std::make_pair(record.getFirstSubfieldValue("150", 'g'), record.getControlNumber()));
         if (not record.getFirstSubfieldValue("150", 'x').empty())
             gnd_keywords->emplace(std::make_pair(record.getFirstSubfieldValue("150", 'x'), record.getControlNumber()));
-
-        const std::string controlNumber = record.getControlNumber();
-        const std::string keywordSubfield = record.getFirstSubfieldValue("150", 'a');
-        const std::string keywordSubfieldG = record.getFirstSubfieldValue("150", 'g');
-        const std::string keywordSubfieldX = record.getFirstSubfieldValue("150", 'x');
-
-       // if (not keywordSubfield.empty() or not keywordSubfieldG.empty() or not keywordSubfieldX.empty())
-       //     std::cout << controlNumber << " a: " << keywordSubfield << " g:  " << keywordSubfieldG << " x: " << keywordSubfieldX << "\n";
-
     }
 
-        LOG_INFO("Processed " + std::to_string(record_count) + " MARC record(s).");
+    LOG_INFO("Processed " + std::to_string(record_count) + " entries.");
     
 }
 
-void findEquivalentKeywords(std::unordered_map<std::string, std::string> keywords_to_gnd, std::set<std::string> keywords_to_compare) {
-    std::map<std::string, std::string> keyword_matches;
-    std::set<std::string> keywords_without_match;
-    std::unordered_map<std::string, std::string>::iterator lookup;
-    for(const auto &keyword : keywords_to_compare) {
-        lookup = keywords_to_gnd.find(keyword);
-        if (lookup != keywords_to_gnd.end()) {
+
+void FindEquivalentKeywords(std::unordered_map<std::string, std::string> const *keywords_to_gnd, std::set<std::string> const *keywords_to_compare, const std::string matches_output_file, const std::string no_matches_output_file) {
+    std::unordered_map<std::string, std::string> keyword_matches;
+    std::unordered_set<std::string> keywords_without_match;
+    //std::unordered_map<std::string, std::string>::iterator lookup;
+    for (const auto &keyword : *keywords_to_compare) {
+        //std::cout << "'" << StringUtil::TrimWhite(keyword) << "'" << "\n";
+        const auto lookup(keywords_to_gnd->find(keyword));
+        //lookup = keywords_to_gnd.find(keyword);
+        if (lookup != keywords_to_gnd->end()) {
             keyword_matches.insert(*lookup);
             continue;
         }
         keywords_without_match.insert(keyword);
     }
     std::cout << "Found: " << keyword_matches.size() << " matches.\n";
+    float percentage = ((float)(keyword_matches.size()*100))/keywords_to_compare->size();
```

Comment:
It is safer to use double unless there is a performance or storage size problem.  Also we put spaces around binary arithmetic operators.  I would use the double(keyword_matches.size() * 100) syntax. (No additional brackets around the keyword.)